static/panes/compiler.ts

```diff
@@ -3289,15 +3294,18 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
         return result;
     }
 
-    private async getAsmInfo(opcode: string, instructionSet: string): Promise<AssemblyInstructionInfo | undefined> {
-        const cacheName = 'asm/' + (instructionSet ? instructionSet + '/' : '') + opcode;
+    private async getAsmInfo(
+        opcode: string,
+        instructionSet: AssemblyDocumentationInstructionSet
+    ): Promise<AssemblyInstructionInfo | undefined> {
+        const cacheName = `asm/${instructionSet}/${opcode}`;
         const cached = OpcodeCache.get(cacheName);
         if (cached) {
             if (cached.found) return cached.data;
             throw new Error(cached.data);
         }
 
-        const response = await getAssemblyDocumentation({opcode: opcode, instructionSet: instructionSet as any});
+        const response = await getAssemblyDocumentation({opcode, instructionSet});
```

Comment:
My solution here is to type the parameter as the instruction set union and let the caller do the cast